src/data/set/intervals/interval.lean

```diff
@@ -0,0 +1,154 @@
+/-
+Copyright (c) 2020 Zhouhang Zhou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Zhouhang Zhou
+-/
+
+import data.set.intervals
+
+/-!
+# Intervals
+
+`a..b` is the set of elements lying between `a` and `b`, with both endpoints included.
+
+`Icc a b` requires the assumption `a ≤ b` to be meaningful, which is sometimes inconvenient. This
+version of interval is always the set of things lying between `a` and `b`, regardless of the
+relative order of `a` and `b`.
+-/
+
+universe u
+
+namespace set
+
+section decidable_linear_order
+
+variables {α : Type u} [decidable_linear_order α] {a a₁ a₂ b b₁ b₂ x : α}
+
+/-- `interval a b` is the set of elements lying between `a` and `b`, with `a` and `b` included. -/
+def interval (a b : α) := if a ≤ b then Icc a b else Icc b a
```

Comment:
Would it be easier to prove lemmas with `interval a b = Icc (min a b) (max a b)`?